storage/remote/queue_manager.go

```diff
@@ -731,17 +678,14 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 
 		if err == nil {
 			succeededSamplesTotal.WithLabelValues(s.qm.queueName).Add(float64(len(samples)))
-			now := time.Now()
-			s.qm.setLastSendTimestamp(now)
-			s.qm.setHighestSentTimestamp(highest)
+			s.qm.highestSentTimestampMetric.Set(float64(highest / 1000))
```

Comment:
are you sure we don't want to check that highest > the value we last set for `highestSentTimestampMetric` first?

Reply:
ah I see, `maxGauge` :+1: